services/app-api/handlers/reports/create.ts

```diff
@@ -127,28 +128,44 @@ export const createReport = handler(
     };
 
     const currentDate = Date.now();
-    let reportYear: number =
-      reportType === ReportType.WP
-        ? new Date(convertDateUtcToEt(currentDate)).getFullYear()
-        : workPlanMetadata!.reportYear;
-
-    let reportPeriod: number =
-      reportType === ReportType.WP
-        ? calculatePeriod(currentDate, workPlanMetadata)
-        : workPlanMetadata!.reportPeriod;
 
     const overrideCopyOver =
       unvalidatedMetadata?.copyReport &&
       unvalidatedMetadata?.copyReport?.isCopyOverTest;
+    let reportYear, reportPeriod;
 
-    if (overrideCopyOver) {
-      if (unvalidatedMetadata?.copyReport?.reportPeriod)
-        reportPeriod = unvalidatedMetadata?.copyReport?.reportPeriod;
-      if (unvalidatedMetadata?.copyReport?.reportYear)
-        reportYear = unvalidatedMetadata?.copyReport?.reportYear;
+    /**
+     * If the report is a WP, determine reportYear from the unvalidated metadata. Otherwise, a SAR will use the workplan metadata.
+     */
+    if (reportType === ReportType.WP) {
+      reportYear = getReportYear(
+        reportType,
+        unvalidatedMetadata,
+        overrideCopyOver
+      );
+    } else {
+      reportYear = getReportYear(
+        reportType,
+        workPlanMetadata,
+        overrideCopyOver
+      );
+    }
 
-      reportYear = reportPeriod == 2 ? reportYear + 1 : reportYear;
-      reportPeriod = (reportPeriod % 2) + 1;
+    /**
+     * If the report is a WP, determine reportPeriod from the unvalidated metadata. Otherwise, a SAR will use the workplan metadata.
+     */
+    if (reportType === ReportType.WP) {
+      reportPeriod = getReportPeriod(
+        reportType,
+        unvalidatedMetadata,
+        overrideCopyOver
+      );
+    } else {
+      reportPeriod = getReportPeriod(
+        reportType,
+        workPlanMetadata,
+        overrideCopyOver
+      );
```

Comment:
duplicate code?

Reply:
Actually passing in different args here (`unvalidatedMetadata` vs `workPlanMetadata`), though now that it's pointed out and I'm looking with fresh eyes, there is definitely a way to make this more elegant, so I will do that.

Reply:
oh yes i see the difference within the `if`. but lines 140-152 are the same as 157-168, no?